statsmodels/graphics/functional.py

```diff
@@ -1,14 +1,441 @@
+                Index value of the components to compute
+        The tuple contains:
+            band : list of float
+                PDF values `[min_pdf, max_pdf]` to be within.
+            pca : statsmodels Principal Component Analysis instance
+                The PCA object to use.
+            bounds : sequence
+                ``(min, max)`` pair for each components
+            ks_gaussian : KDEMultivariate instance
+
+    Returns
+    -------
+    band : tuple of float
+        ``(max, min)`` curve values at `idx`
+
+    """
+    idx, (band, pca, bounds, ks_gaussian) = args
+    if have_de_optim:
+        max_ = differential_evolution(_curve_constrain, bounds=bounds,
+                                      args=(idx, -1, band, pca, ks_gaussian),
+                                      maxiter=7).x
+        min_ = differential_evolution(_curve_constrain, bounds=bounds,
+                                      args=(idx, 1, band, pca, ks_gaussian),
+                                      maxiter=7).x
+    else:
+        max_ = brute(_curve_constrain, ranges=bounds, finish=fmin,
+                     args=(idx, -1, band, pca, ks_gaussian))
+
+        min_ = brute(_curve_constrain, ranges=bounds, finish=fmin,
+                     args=(idx, 1, band, pca, ks_gaussian))
+
+    band = (_inverse_transform(pca, max_)[0][idx],
+            _inverse_transform(pca, min_)[0][idx])
+    return band
+
+
+def hdrboxplot(data, ncomp=2, alpha=None, threshold=0.95, bw=None,
+               xdata=None, labels=None, ax=None):
+    """Plot High Density Region boxplot.
+
+    1. Compute a multivariate kernel density estimation,
+    2. Compute contour lines for quantiles 90%, 50% and `alpha`%,
+    3. Plot the bivariate plot,
+    4. Compute mediane curve along with quantiles and outliers curves.
+
+    Parameters
+    ----------
+    data : sequence of ndarrays or 2-D ndarray
+        The vectors of functions to create a functional boxplot from.  If a
+        sequence of 1-D arrays, these should all be the same size.
+        The first axis is the function index, the second axis the one along
+        which the function is defined.  So ``data[0, :]`` is the first
+        functional curve.
+    ncomp : int, optional
+        Number of components to use.  If None, returns the as many as the
+        smaller of the number of rows or columns in data.
+    alpha : list of floats between 0 and 1, optional
+        Extra quantile values to compute. Default is None
+    threshold : float between 0 and 1
+        Percentile threshold value for outliers detection. High value means
+        a lower sensitivity to outliers. Default is `0.95`.
+    bw: array_like or str, optional
+        If an array, it is a fixed user-specified bandwidth.  If a string,
+        should be one of:
+
+            - normal_reference: normal reference rule of thumb (default)
+            - cv_ml: cross validation maximum likelihood
+            - cv_ls: cross validation least squares
```

Comment:
this should also specify what the default is, if bw=None

Reply:
Added some clarification